app/src/main/java/com/kickstarter/ui/fragments/DiscoveryFragment.kt

```diff
@@ -47,34 +49,41 @@ import com.kickstarter.ui.data.Editorial
 import com.kickstarter.ui.data.LoginReason
 import com.kickstarter.ui.viewholders.EditorialViewHolder
 import com.kickstarter.viewmodels.DiscoveryFragmentViewModel
-import rx.android.schedulers.AndroidSchedulers
-import rx.schedulers.Schedulers
+import com.trello.rxlifecycle.FragmentEvent
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.schedulers.Schedulers
+import kotlinx.coroutines.launch
 
-@RequiresFragmentViewModel(DiscoveryFragmentViewModel.ViewModel::class)
-class DiscoveryFragment : BaseFragment<DiscoveryFragmentViewModel.ViewModel>() {
+class DiscoveryFragment : Fragment() {
     private var heartsAnimation: AnimatorSet? = null
-    private var recyclerViewPaginator: RecyclerViewPaginator? = null
+    private var recyclerViewPaginator: RecyclerViewPaginatorV2? = null
 
     private var binding: FragmentDiscoveryBinding? = null
     private var discoveryEditorialAdapter: DiscoveryEditorialAdapter? = null
     private val projectStarConfirmationString = R.string.project_star_confirmation
+    private val disposables = CompositeDisposable()
+
+    private lateinit var viewModelFactory: DiscoveryFragmentViewModel.Factory
+    private val viewModel: DiscoveryFragmentViewModel.DiscoveryFragmentViewModel by viewModels { viewModelFactory }
 
     override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
         super.onCreateView(inflater, container, savedInstanceState)
+        this.context?.getEnvironment()?.let { env ->
+            viewModelFactory = DiscoveryFragmentViewModel.Factory(env)
+        }
         binding = FragmentDiscoveryBinding.inflate(inflater, container, false)
         return binding?.root
     }
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
 
-        this.lifecycle()
-            .compose(bindToLifecycle())
-            .compose(Transformers.observeForUI())
-            .filter { it.isNotNull() }
-            .subscribe {
-                this.viewModel.inputs.fragmentLifeCycle(it)
+        viewLifecycleOwner.lifecycleScope.launch {
+            repeatOnLifecycle(Lifecycle.State.RESUMED) {
+                activity?.runOnUiThread { viewModel.inputs.fragmentLifeCycle(FragmentEvent.RESUME) }
```

Comment:
nitpick: `Fragment.RESUME` enum belongs to `com.trello.rxlifecycle` library that we want to get rid off alongside RXJava1.
We could here change the type that `viewModel.inputs.fragmentLifecyle` receives from `com.trello.rxlifecycle.FragmentEvent.Lifecycle.State` to `androidx.lifecycle.Lifecycle.State`